sdk/containerregistry/Microsoft.Azure.ContainerRegistry/src/Customizations/AcrClientCredentials.cs

```diff
@@ -153,40 +160,55 @@ public override async Task ProcessHttpRequestAsync(HttpRequestMessage request, C
             }
             else
             {
-                string operation = "https://" + _loginUrl + request.RequestUri.AbsolutePath;
-                string scope = await getScope(operation, request.Method.Method, request.RequestUri.AbsolutePath);
+                string operation = $"https://{_loginUrl}{request.RequestUri.AbsolutePath}";
+                string scope = await GetScope(operation, request.Method.Method, request.RequestUri.AbsolutePath);
 
-                request.Headers.TryAddWithoutValidation("Authorization", "Bearer " + getAcrAccessToken(scope));
+                request.Headers.TryAddWithoutValidation("Authorization", $"Bearer {GetAcrAccessToken(scope)}");
             }
-            await base.ProcessHttpRequestAsync(request, cancellationToken);
 
+            await base.ProcessHttpRequestAsync(request, cancellationToken);
         }
 
         #endregion
 
         #region Helpers
 
+        private static string ProcessLoginUrl(string loginUrl)
+        {
+            // Proofing in case passed in loginurl includes https start.
+            if (loginUrl.ToLower().StartsWith("https://"))
```

Comment:
Does it make sense to add a `Debug.Assert(!loginUrl.StartsWith("http://"))` to make sure you never use non https?